core/src/test/scala/fs2/TestUtil.scala

```diff
@@ -32,7 +34,7 @@ trait TestUtil {
     catch {
       case e: InterruptedException => throw e
       case e: TimeoutException => throw e
-      case e: Throwable => ()
+      case NonFatal(e) => ()
```

Comment:
Hm, I am just thinking loudly. In strategies and executors shouldn't we just do System.exit(-1) on all fatal exceptions after stderr them? As I am aware only exception that JVM may terminate is OOM (that we perhaps shouldn't catch at all ever), all the rest will lead us to unpredictable state. Rather would kill JVM then having it run in un-predicatable state.